chartfx-chart/src/test/java/de/gsi/chart/plugins/ScreenshotTest.java

```diff
@@ -0,0 +1,154 @@
+        chart.setPrefHeight(600);
+        chart.getDatasets().add(new CosineFunction("Cosine", 50));
+        stage.setScene(scene);
+        stage.show();
+    }
+
+    @Test
+    void screenshotTests() throws IOException {
+        // setup some useful variables
+        final FlowPane toolbar = chart.getToolBar();
+        Path tmpdir = Files.createTempDirectory("screenshotTest");
+
+        // add Screenshot Plugin
+        fxRobot.interact(() -> {
+            chart.getPlugins().add(screenshotPlugin);
+        });
+
+        // set and test Properties
+        fxRobot.interact(() -> {
+            // pattern
+            assertEquals("", screenshotPlugin.getPattern());
+            screenshotPlugin.setPattern("{dataSetName}_{xMin}.png");
+            assertEquals("{dataSetName}_{xMin}.png", screenshotPlugin.getPattern());
+            // directory
+            screenshotPlugin.setDirectory(tmpdir.toAbsolutePath().toString());
+            assertEquals(tmpdir.toAbsolutePath().toString(), screenshotPlugin.getDirectory());
+        });
+
+        // verify that controls were added to toolbar
+        FxAssert.verifyThat(toolbar, node -> node.getChildren().size() == 1);
+
+        // check dimensions of screenshot match chart dimensions
+        fxRobot.interact(() -> {
+            final Image screenshot = screenshotPlugin.getScreenshot();
+            assertEquals(chart.getWidth(), screenshot.getWidth());
+            assertEquals(chart.getHeight(), screenshot.getHeight());
+        });
+
+        // check copying of screenshot to clipboard
+        fxRobot.interact(() -> {
+            screenshotPlugin.screenshotToClipboard();
+            final Clipboard clipboard = Clipboard.getSystemClipboard();
+            final Image clipboardContents = (Image) clipboard.getContent(DataFormat.IMAGE);
+            assertNotNull(clipboardContents);
+            assertEquals(chart.getWidth(), clipboardContents.getWidth());
+            assertEquals(chart.getHeight(), clipboardContents.getHeight());
+        });
+
+        // check saving screenshot to file
+        fxRobot.interact(() -> screenshotPlugin.screenshotToFile(false));
+
+        File[] filelist = tmpdir.toAbsolutePath().toFile().listFiles();
+        assertEquals(1, filelist.length);
+        assertEquals(new File(tmpdir.toAbsolutePath().toString(), "Cosine_-0.1.png"), filelist[0]);
+        WritableImage imageRecovered = SwingFXUtils.toFXImage(ImageIO.read(filelist[0]), null);
+        fxRobot.interact(() -> {
+            assertEquals(chart.getWidth(), imageRecovered.getWidth());
+            assertEquals(chart.getHeight(), imageRecovered.getHeight());
+        });
+
+        // remove plugin
+        fxRobot.interact(() -> chart.getPlugins().remove(screenshotPlugin));
+
+        // verify that controls were removed from toolbar
+        FxAssert.verifyThat(toolbar, node -> node.getChildren().size() == 0);
+    }
+
+    @Test
+    void settingsDialogTests() {
+
+    }
+
+    @Test
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5084368)